addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/internal/core/LxJsonApp3.java

```diff
@@ -21,9 +23,9 @@
 class LxJsonApp3 {
 
     LxJsonInfo msInfo;
-    Map<String, LxJsonControl> controls;
-    Map<String, LxJsonRoom> rooms;
-    Map<String, LxJsonCat> cats;
+    Map<@NonNull String, @NonNull LxJsonControl> controls;
```

Comment:
Please remove all @NonNull and use @NonNullByDefault

Reply:
Then I have to add nullable to every field, since they can be null. It also gives me over 20 new warnings due to potential null access, which are these warnings for potential change of the value in multithreaded code, which will never happen here. For example:
```
if (a != null) {
    if (a.b != null) {
        a.b.doSomething();
    }
}
```
This will generate warning at doSomething, because `a` might have changed between `a!=null` and this line.
I think it is not worth it and would rather leave the class not annotated. But then I have to mark the Map key and value and nonnull, because they are never created null, if map is created at all and this seems to break the convention to not use nonnull.
What do you think, which is better? Maybe another solution exists, which I can't see now.

Reply:
I was triggered by the remark of @triller-telekom maybe he knows more.

Reply:
@triller-telekom can you enlighten us?

Reply:
Maybe you want to join in https://github.com/eclipse/smarthome/issues/4321 with these observations/problems. The more we use those annotations, the less we are confident that it is possible to solve all the warnings in a decent way...

Reply:
@kaikreuzer @ppieczul I just read the latest state of the discussion and now I'm in doubt should we just merge this as is. Or do we still want to wait for things to settle down. I think that I read that the usage of @nonnullbydefault is still the prefered way, but not all warnings have to be solved / surpressed?

I think it would be great of one of people actively involved in the discussion could review this PR, maybe @htreu @triller-telekom or @sjka? Thanks all!

Reply:
It would be great to have this reviewed by more people. 

If there are generally no erroneous changes here, I think (in the spirit of continuous integration) it is more beneficial to merge this and then resolve outstanding warnings at the JDK calls - when they are solved at the framework level. 

I will also summarize my findings in the smarthome disscuscion mentioned above.

With more and more nonnullbydefault added to the framework, I am now getting lots of warnings from this binding too with the version that is on master.
Also, there is a new version of Loxone server software which requires new things to be added to the binding, so it starts stacking up.
Thanks

Reply:
I would vote for adding `@NonNullbyDefault` to the class and if your variables can all be `null` you would need to add the `@Nullable` to them.

And please add your findings regarding false positives warnings about the null annotations to our discussion in https://github.com/eclipse/smarthome/issues/4321. The outcome of this discussion will probably be that we change the settings within Eclipse about what we will treat as a warning/error.

For sure we should not block PRs for senseless discussions, but in this case we have the problem that we introduced the null annotations to the framework before we actually agreed on rules on how to use them :( This was a bit unfortunate because while using those annotations they turned out to be rather tricky to use and the tools (like Eclipse) do not support them perfectly. I am sorry that you had to change your PR quite some often because of this, but you were a little to quick for us ;)